tests/test_dictField.py

```diff
@@ -80,6 +80,33 @@ class BadDictCheck(pexConfig.Config):
         else:
             raise SyntaxError("Non-callable dictCheck DictFields should not be allowed")
 
+    def testFieldTypeAnnotationRuntime(self):
+        # test parsing type annotation for runtime keytype, itemtype
+        testField = pexConfig.DictField[str, int](doc="")
```

Comment:
This syntax indeed does not work, because there is actually nothing linking the right side expression to the left (unless as you say `ConfigMeta` gets involved which brings in a host of other issues that I would prefer not to tackle on this ticket.

However, `testField: DictField[str, int]` does work to type the variable in any other context, but it can't inject the  type information.